test/chainnotifier_mock.go

```diff
@@ -103,7 +105,16 @@ func (c *mockChainNotifier) RegisterConfirmationsNtfn(ctx context.Context,
 	txid *chainhash.Hash, pkScript []byte, numConfs, heightHint int32) (
 	chan *chainntnfs.TxConfirmation, chan error, error) {
 
-	confChan := make(chan *chainntnfs.TxConfirmation, 1)
+	reg := &ConfRegistration{
+		PkScript:   pkScript,
+		TxID:       txid,
+		HeightHint: heightHint,
+		NumConfs:   numConfs,
+		ConfChan:   make(chan *chainntnfs.TxConfirmation, 1),
+	}
+
+	c.confRegistrations = append(c.confRegistrations, reg)
```

Comment:
Thread safety?

Reply:
Good catch, I guess I didn't notice this as there weren't concurrent registrations in our current tests, not even after adding native segwit.